bee-network/src/conn.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::HashSet,
+    hash::Hash,
+    net::SocketAddr,
+    sync::{Arc, RwLock},
+};
+
+pub enum Direction {
+    Inbound,
+    Outbound,
+}
+
+pub type ConnectedList = FilterList<SocketAddr>;
+
+#[derive(Clone)]
+pub struct FilterList<T: Eq + Hash>(Arc<RwLock<HashSet<T>>>);
+
+impl<T: Eq + Hash> FilterList<T> {
+    pub fn new() -> Self {
+        Self(Arc::new(RwLock::new(HashSet::new())))
+    }
+
+    pub fn add(&self, item: T) {
+        self.0.write().expect("error getting the lock").insert(item);
+    }
+
+    // TODO: update this list on connection loss
+    pub fn _remove(&self, item: T) -> bool {
```

Comment:
because it's dead code atm, but will be used as soon as we also want to remove some peers. That's not the case atm, because we have no autopeering. I removed the underscore and allowed dead code instead.